libparsec/crates/platform_storage/src/native/certificates.rs

```diff
@@ -183,6 +183,19 @@ impl PlatformCertificatesStorage {
     ) -> anyhow::Result<()> {
         let index_as_i64 = i64::try_from(index)?;
 
+        // TODO: Avoid this check in batch query
```

Comment:
move this in the sql transaction, typically doing a returning:

```sql
INSERT INTO certificates(certificate_index, certificate, ...)
  VALUES (?, ?, ...)
  RETURNING (
    SELECT certificate_index
    FROM certificates
    ORDER BY certificate_index DESC
    LIMIT 1
    OFFSET 1
);
```

This will return the previous maximum value of certificate_index (so equivalent to `get_last_index()`), from this you can check that the returned value equals `new_index - 1`, and if not rollback the transaction by returning an error